src/crafting.cpp

```diff
@@ -60,13 +60,13 @@
 #include "type_id.h"
 #include "clzones.h"
 #include "colony.h"
-#include "faction.h"
 #include "flat_set.h"
 #include "iuse.h"
 #include "point.h"
 #include "weather.h"
 
 class basecamp;
+class faction;
```

Comment:
This is not needed. It's already declared by the included header "player.h" and probably by others as well.